### PR TITLE
User mentions handling should be skipped when no fields were updated

### PR DESCRIPTION
**Pull request by cedric-anne**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -

Edge case revealed by #12451 .

### CODE REVIEW

**Comment by cedric-anne** [comment]
I pushed a patch to ensure existing tests are made in the real GLPI world (the world where quotes are always escaped).

Problem here was that tests were made on a data that was not sanitized, and therefore, where quotes were not backslashed. When update method was call using the exact same content as the content used on creation, it was still detected as a change due to following code: https://github.com/glpi-project/glpi/blob/6ad63544a71e0033a15ddaa42b7602523351b16f/src/CommonDBTM.php#L1622-L1649

So basically, when a string contains a `"`:
```php
$this->fields[$key] != $this->input[$key] // => false
$DB->escape($this->fields[$key]) != $this->input[$key] // => true
```

Running the test without proposed patch is failing as expected:
image
